libraries/AC_Planck/AC_Planck.cpp

```diff
@@ -286,3 +333,76 @@ bool AC_Planck::get_posvel_cmd(Location &loc, Vector3f &vel_cms, float &yaw_cd,
   _cmd.is_new = false;
   return true;
 }
+
+bool AC_Planck::check_if_high_tension_failed(const int32_t alt_cm) {
+  //No comms from the tether
+  bool tether_comms_failed = is_tether_timed_out();
+
+  //Determine if high tension has been or should have been triggered
+  bool high_tension_triggered = _tether_status.high_tension || tether_comms_failed;
+
+  //Hasn't failed if not triggered
+  if(!high_tension_triggered) {
+    _tether_status.sent_failed_message = false;
+    return false;
+  }
+
+  //Check for a timeout. Has it been 15s since the tether indicated high tension,
+  //or 15s since we last heard the tether comms timed out
+  bool timeout = false;
+  if(_tether_status.high_tension) {
+    timeout = (AP_HAL::millis() - _tether_status.high_tension_timestamp_ms) > 15000;
+  } else { //Tether comms have failed
+    timeout = (AP_HAL::millis() - _tether_status.timestamp_ms) > 20000; //15s + 5s timeout
+  }
+
+  //If no timeout, it hasn't failed yet
+  if(!timeout) {
+    _tether_status.sent_failed_message = false;
+    return false;
+  }
+
+  //Check if the altitude stayed the same or increased. Add a 2m buffer
+  float alt_threshold = _tether_status.high_tension_alt_cm - 200.;
+  bool alt_increased = ((float)alt_cm >= alt_threshold);
+
+  //Check if the tag altitude stayed the same or increased. Add a 2m buffer
```

Comment:
Duplicate comment

Reply:
Its not. This is the check on tag altitude, not EKF altitude

Reply:
Oh, whoops, missed the "tag", nevermind